web/app/scripts/navbar/navbar-controller.js

```diff
@@ -0,0 +1,60 @@
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function NavbarController($log, $rootScope, $state, $stateParams, RecordTypes) {
+        var ctl = this;
+        ctl.updateState = updateState;
+
+        setRecordTypes();
+        setStates();
+        $rootScope.$on('$stateChangeSuccess', setStates);
+
+        // Sets the list of available record types, and also the active one if provided
+        function setRecordTypes() {
+            RecordTypes.query({ active: 'True' })
+                .$promise.then(function(recordTypes) {
```

Comment:
I think this could be written more concisely as `RecordTypes.query({ active: 'True'}, function(...) {})` but I'm not sure what our preferred style is; it's kind of nice making the promise semantics explicit.